From 612e309bfffd6f9b8ad7cdccda3019fc0865f3b4 Mon Sep 17 00:00:00 2001
From: Hans Zandbelt <hzandbelt@pingidentity.com>
Date: Thu, 19 Jan 2017 00:03:37 +0100
Subject: [PATCH] don't echo query params on invalid requests to redirect URI;
 closes #212

thanks @LukasReschke; I'm sure there's some OWASP guideline that warns
against this
---
 ChangeLog              | 3 +++
 src/mod_auth_openidc.c | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 4010fb39..b046b8d5 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,6 @@
+01/18/2017
+- don't echo the query parameters on the error page when an invalid request is made to the Redirect URI; closes #212; thanks @LukasReschke
+
 01/14/2017
 - use dynamic memory buffer for writing HTTP call responses; solves curl/mpm-event interference; see #207
 - bump to 2.1.4rc1
diff --git a/src/mod_auth_openidc.c b/src/mod_auth_openidc.c
index fea57754..1a2fc55f 100644
--- a/src/mod_auth_openidc.c
+++ b/src/mod_auth_openidc.c
@@ -2845,8 +2845,8 @@ int oidc_handle_redirect_uri_request(request_rec *r, oidc_cfg *c,
 	/* something went wrong */
 	return oidc_util_html_send_error(r, c->error_template, "Invalid Request",
 			apr_psprintf(r->pool,
-					"The OpenID Connect callback URL received an invalid request: %s",
-					r->args), HTTP_INTERNAL_SERVER_ERROR);
+					"The OpenID Connect callback URL received an invalid request"),
+					HTTP_INTERNAL_SERVER_ERROR);
 }
 
 /*
